Docs/describe debug levels (#1200)
reimagined/resolve: docs/debugging.md

---
id: debugging
title: Debugging
---

> Resolve uses the [resolve-debug-levels](https://www.npmjs.com/package/resolve-debug-levels) package to log debug information. This package extends the [debug](https://www.npmjs.com/package/debug) library's functionality with logging levels. The package provides the following logging levels:

Reviewer: ```suggestion
Resolve uses the [resolve-debug-levels](https://www.npmjs.com/package/resolve-debug-levels) package to log debug information. This package extends the [debug](https://www.npmjs.com/package/debug) library's functionality with the following logging levels:
```

| Level   | Description                                            |
| ------- | ------------------------------------------------------ |
| log     | Messages that should always be displayed               |

> | error   | Errors that can hinder normal program execution        |

Reviewer: ```suggestion
| error   | Errors that can prevent normal program execution        |
```

| warn    | Potential problems in the application's implementation |
| debug   | Information displayed for debugging purposes           |
| info    | Messages that describe the current operation           |

> | verbose | Extended descriptions to the previous message          |

Reviewer: "to the previous message" Why 'to'?

Author: Changed to 'Extended descriptions for the previous message'

Reviewer: Ok

Use the `DEBUG_LEVEL` environment variable to specify the logging level.

## Debug ReSolve

The reSolve framework uses the `resolve` prefix for all its debugging namespaces. To enable the framework's debug output, assign `resolve*` to the `DEBUG` environment variable as shown below:

```
DEBUG=resolve:* DEBUG_LEVEL=error yarn dev
```

> The reSolve framework displays messages of the **log**, **error** and **warn** levels by default.

Reviewer: ```suggestion
The reSolve framework displays **log**, **error**, and **warn** level messages.
```

## Debug a ReSolve Application

> You can use [resolve-debug-levels](https://www.npmjs.com/package/resolve-debug-levels) to debug your reSolve application. To do this, add the `resolve-debug-levels` package to your application's dependencies:

Reviewer: ```suggestion
You can add the `resolve-debug-levels` package to your application's dependencies to use [resolve-debug-levels](https://www.npmjs.com/package/resolve-debug-levels) to debug your reSolve application.
```

```
yarn add resolve-debug-levels
```

> To create a logger, pass your module's debugging namespace to the function exposed by the `resolve-debug-levels` module:

Reviewer: ```suggestion
To create a logger, pass your module's debugging namespace to the function the `resolve-debug-levels` module exposes:
```

```js
import debugLevels from 'resolve-debug-levels'
const log = debugLevels('myapp:api-handlers')
...
```

> The logger object exposes methods that correspond to the available debug levels:

Reviewer: "to the available debug" Why 'the available'?

Author: changed to "to the debug levels"

Reviewer: Ok
